cmd/otel-allocator/server/server.go

```diff
@@ -42,27 +41,49 @@ var (
 	}, []string{"path"})
 )
 
+var (
+	yamlConfig = jsoniter.Config{
+		EscapeHTML:                    false,
+		MarshalFloatWith6Digits:       true,
+		ObjectFieldMustBeSimpleString: true,
+		TagKey:                        "yaml",
+	}.Froze()
+	jsonConfig = jsoniter.Config{
+		EscapeHTML:                    false,
+		MarshalFloatWith6Digits:       true,
+		ObjectFieldMustBeSimpleString: true,
+	}.Froze()
+)
+
 type collectorJSON struct {
 	Link string         `json:"_link"`
 	Jobs []*target.Item `json:"targets"`
 }
 
+type DiscoveryManager interface {
+	GetScrapeConfigs() map[string]*promconfig.ScrapeConfig
+}
+
 type Server struct {
 	logger           logr.Logger
 	allocator        allocation.Allocator
-	discoveryManager *target.Discoverer
+	discoveryManager DiscoveryManager
 	server           *http.Server
+	yamlMarshaller   jsoniter.API // TODO: for testing, it would make a lot of sense to have a simpler interface
+	jsonMarshaller   jsoniter.API // TODO: for testing, it would make a lot of sense to have a simpler interface
```

Comment:
@jaronoff97, do you think I should define my own interfaces here or keep using the `jsoniter.API` interface (which is pretty big)?

Reply:
Is there a more generic interface these adhere to?

Reply:
Not really 😕